include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -153,14 +153,13 @@ SpatialOperatorBase<dim, Number>::fill_matrix_free_data(
                                      field + quad_index_p);
   matrix_free_data.insert_quadrature(dealii::QGauss<1>(param.degree_u + (param.degree_u + 2) / 2),
                                      field + quad_index_u_nonlinear);
-  if(param.store_previous_boundary_values)
-  {
-    matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.degree_u + 1),
-                                       field + quad_index_u_gauss_lobatto);
-    matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.get_degree_p(param.degree_u) +
-                                                                1),
-                                       field + quad_index_p_gauss_lobatto);
-  }
+
+  // TODO create those quadrature rules only when needed
+  matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.degree_u + 1),
+                                     field + quad_index_u_gauss_lobatto);
+  matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.get_degree_p(param.degree_u) +
+                                                              1),
+                                     field + quad_index_p_gauss_lobatto);
```

Comment:
To enable an additional quadrature formula implies a considerable increase in memory consumption on general deformed meshes. In the longer run, we need to equip `FEEvaluation` with an entry point where we can compute the mapping information on the fly. Here I am not talking about the possible performance benefit (for which the current interfaces of deal.II are likely inappropriate as they force us into too much overhead), but really just the memory overhead for rarely used functionality.

Reply:
For sure, this is nothing we need to adjust in this PR, but we need to work towards a solution avoiding excessive memory consumption of `MatrixFree` when we need new evaluation points.